pkg/kubelet/pod_workers.go

```diff
@@ -537,6 +537,14 @@ func (p *podWorkers) UpdatePod(options UpdatePodOptions) {
 	var wasGracePeriodShortened bool
 	switch {
 	case status.IsTerminated():
+		// A terminated pod may still be waiting for cleanup - if we recieve a runtime pod kill request
+		// due to housekeeping seeing an older cached version of the runtime pod simply ignore it until
+		// after the pod worker completes.
+		if isRuntimePod {
+			klog.InfoS("Pod is waiting for termination, ignoring runtime-only kill until after pod worker is fully terminated", "pod", klog.KObj(pod), "podUID", pod.UID)
```

Comment:
Once we validate this works, this log line should be at level 4 or so.

Reply:
This should be so rare that I'd prefer to have it at 1, possibly at 2.

Reply:
Set it at 3 (in between).